Realm/RLMArrayLinkView.mm

```diff
@@ -194,15 +200,11 @@ - (void)replaceObjectAtIndex:(NSUInteger)index withObject:(RLMObject *)object {
 - (NSUInteger)indexOfObject:(RLMObject *)object {
     // check attached for table and object
     RLMLinkViewArrayValidateAttached(self);
+
     if (object->_realm && !object->_row.is_attached()) {
         @throw RLMException(@"RLMObject is no longer valid");
     }
 
-    // check that object types align
```

Comment:
We need to re-add the object check, and add a test for calling this method with an RLMObject from a different realm.